src/scripts/main.js

```diff
@@ -1 +1,63 @@
+
 'use strict';
+
+function firstPromise() {
+  return new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  });
+}
+
+function secondPromise() {
+  return new Promise(resolve => {
+    document.addEventListener('mousedown', (e) => {
+      if (e.button === 0 || e.button === 2) {
+        resolve('Second promise was resolved');
+      }
+    });
+  });
+}
+
+function thirdPromise() {
+  return new Promise(resolve => {
+    let leftBtn = false;
+    let rightBtn = false;
+
+    document.addEventListener('mousedown', (e) => {
+      switch (e.button) {
+        case 0:
+          leftBtn = true;
+          break;
+        case 2:
+          rightBtn = true;
+          break;
+        default:
+          throw new Error('Click left or right button');
```

Comment:
Don't throw unnecessary errors, especially when they are not handled.  Either exclude the default case from switch or just do a regular if statement with || operator